### PR TITLE
linux: enable CR50 TPM found on chromebooks

### PR DESCRIPTION
**Pull request by jmbaur**
###### Description of changes

Along with other ChromeOS related configs, this change enables TPM support for Chromebook/Chromebox devices.

@zhaofengli pinging you since it seems you added the other related options.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/os-specific/linux/kernel/common-config.nix [hunk 1]
```diff
@@ -1023,6 +1023,8 @@ let
       CROS_EC_ISHTP = module;
 
       CROS_KBD_LED_BACKLIGHT = module;
+
+      TCG_TIS_SPI_CR50 = yes;
```
Comment (Tungsten842) [inline pkgs/os-specific/linux/kernel/common-config.nix:1027]:
```suggestion
      TCG_TIS_SPI_CR50 = whenAtLeast "5.5" yes;
```
https://cateee.net/lkddb/web-lkddb/TCG_TIS_SPI_CR50.html

Comment (jmbaur) [reply pkgs/os-specific/linux/kernel/common-config.nix]: Thank you!

**Review by alyssais — APPROVED** [review]
To get this merged, there are two choices:

- Wait for the next round of stable kernel updates, and ask on that PR to have this one merged at the same time so we can group the rebuilds together.
- Change the base branch to staging, where we can merge straight away, but then you'll have to wait a staging cycle before it's available.

Up to you — depends on whether you feel like watching the PR list to pounce on a stable kernels PR.

**Comment by alyssais** [comment]
(small note: generally, you should use the attribute name as the commit message prefix, so "linux:" rather than "kernel:")

**Comment by alyssais** [comment]
@OfBorg build linux_5_10 linux_5_15 linux_6_1 linux_latest

**Pull request merged**